src/GitHubApps/src/DarcBot/AzureFunction.cs

```diff
@@ -0,0 +1,212 @@
+            [HttpTrigger(AuthorizationLevel.Anonymous, "POST", Route = null)]
+            HttpRequestMessage req, ILogger log)
+        {
+            _log = log;
+
+            dynamic data = await req.Content.ReadAsAsync<object>();
+
+            string userType = data?.comment?.user?.type;
+            if (userType == "Bot")
+            {
+                _log.LogInformation("Comment is from DarcBot, ignoring.");
+                return new OkObjectResult($"Ignoring DarcBot comment");
+            }
+            string gitHubAction = data?.action;
+
+            int installationId = data?.installation?.id;
+            long repositoryId = data?.repository?.id;
+            int prNumber = data?.issue?.number;
+
+            // Check for darcbot directed comments
+            if (gitHubAction == "created")
+            {
+                /* Nothing to do, this is the section where you would handle user comments if they contained explicit directives to darcbot.
+                   Note: be sure to ignore comments from issue?.user?.type == "Bot" so you don't get stuck in a payload delivery loop */
+            }
+
+            if (gitHubAction != "opened" &&
+                gitHubAction != "reopened" &&
+                gitHubAction != "closed")
+            {
+                _log.LogInformation($"Received github action '{gitHubAction}', nothing to do");
+                return new OkObjectResult($"DarcBot has nothing to do with github issue action '{gitHubAction}'");
+            }
+
+            // Determine identifiable information for triage item
+            string issueBody = data?.issue?.body;
+            TriageItem triageItem = GetTriageItemProperties(issueBody);
+
+            string issueUrl = data?.issue?.html_url;
+            triageItem.Url = issueUrl;
+
+            if (triageItem == null)
+            {
+                /* Item is not a triage item (does not contain identifiable information), do nothing */
+                _log.LogInformation($"{data?.issue?.url} is not a triage type issue.");
+                return new OkObjectResult("No identifiable information detected");
+            }
+
+            IGitHubClient gitHubClient = new DarcBotGitHubClient(installationId, _log);
+
+            if (gitHubAction == "opened" ||
+                gitHubAction == "reopened")
+            {
+                
+                // First, look for duplicate issues that are open
+                var openIssues = new RepositoryIssueRequest
+                {
+                    Filter = IssueFilter.All,
+                    State = ItemStateFilter.Open,
+                    SortProperty = IssueSort.Created,
+                    SortDirection = SortDirection.Ascending
+                };
+                _log.LogInformation("Getting open issues");
+                var issues = gitHubClient.Issue.GetAllForRepository(repositoryId, openIssues).Result;
+                _log.LogInformation("Acquired open issues");
```

Comment:
We can probably combine these logs statements. :-) Save ourselves a write to app insights (or wherever these logs go)...

Reply:
Yes, they go to app insights.  Are the writes to app insights problematic?  I can delete most of them, I just left them because they were useful during development so I figured they may be useful to someone else in the future.

Reply:
Mostly it's just when there are a bunch in a row with no logic between them.  Each individual log technically costs $, so combining them when possible is ideal.